### PR TITLE
[NCL-4999] implemented test for EnvironmentProvider

### PR DESCRIPTION
**Pull request by michalovjan**
### Checklist:

* [x] Have you added unit tests for your change?

### CODE REVIEW

File: facade/src/test/java/org/jboss/pnc/facade/providers/AbstractProviderTest.java [hunk 1]
```diff
@@ -162,6 +172,15 @@ public void prepareRepositoryMock() {
...
         });
+        doAnswer(inv -> {
+            Integer id = inv.getArgument(0);
+            Object object = repositoryList.stream()
+                    .filter(a -> id.equals(a.getId()))
+                    .findFirst()
+                    .orElse(null);
+            repositoryList.remove(object);
+            return null;
+        }).when(repository()).delete(anyInt());
```
Comment (janinko) [inline facade/src/test/java/org/jboss/pnc/facade/providers/AbstractProviderTest.java]: Why do you use it in different order then the previous things are used?

File: facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    public void testGetAll() {
+        Page<Environment> all = provider.getAll(0, 10, null, null);
+
+        assertThat(all.getContent())
+                .hasSize(4)
+                .haveExactly(1, new Condition<>(e -> e.getName().equals(env.getName()), "Environment present"));
+    }
+
+    @Test
+    public void testStore() {
```
Comment (janinko) [inline facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java]: I think we don't allow storing of environments via REST.

File: facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    @Test
+    public void testGetSpecific() {
+        Environment environment = provider.getSpecific(env.getId().toString());
+
+        assertThat(environment.getId()).isEqualTo(env.getId().toString());
+        assertThat(environment.getName()).isEqualTo(env.getName());
+    }
+
+    @Test
+    public void testStoreWithId(){
```
Comment (janinko) [inline facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java]: I think we don't allow storing of environments via REST.

File: facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        try{
+            provider.store(environment);
+            fail("Validation exception expected");
+        }catch(DTOValidationException ex){
+            // ok
+        }
+    }
+
+    @Test
+    public void testUpdate(){
```
Comment (janinko) [inline facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java]: I think we don't allow updating of environments via REST.

File: facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+        Environment updated = provider.update(env.getId().toString(), environment);
+
+        assertThat(updated.getId()).isEqualTo(env.getId().toString());
+        assertThat(updated.getName()).isNotEqualTo(env.getName());
+        assertThat(updated.getSystemImageType()).isEqualTo(env.getSystemImageType());
+        assertThat(updated.getSystemImageId()).isEqualTo(env.getSystemImageId());
+    }
+
+    @Test
+    public void testDelete(){
```
Comment (janinko) [inline facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java]: I think we don't allow deleting of environments via REST.

File: facade/src/test/java/org/jboss/pnc/facade/providers/AbstractProviderTest.java (hunk 1 above)
Comment (michalovjan) [reply facade/src/test/java/org/jboss/pnc/facade/providers/AbstractProviderTest.java]: Delete returns void and there was a problem to use it inside when() method.

File: facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java (hunk 2 above)
Comment (michalovjan) [reply facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java]: No, but other providers may use it I guess?

File: facade/src/test/java/org/jboss/pnc/facade/providers/AbstractProviderTest.java (hunk 1 above)
Comment (janinko) [reply facade/src/test/java/org/jboss/pnc/facade/providers/AbstractProviderTest.java]: right

File: facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java (hunk 2 above)
Comment (michalovjan) [reply facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java]: I'll update the provider to UnsupportedOpException

File: facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java (hunk 3 above)
Comment (michalovjan) [reply facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java]: YEP

File: facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java (hunk 4 above)
Comment (michalovjan) [reply facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java]: YEP

File: facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java (hunk 5 above)
Comment (michalovjan) [reply facade/src/test/java/org/jboss/pnc/facade/providers/EnvironmentProviderTest.java]: YEP

**Comment by thescouser89** [comment]
retest this please

**Comment by michalovjan** [comment]
retest this please

**Comment by michalovjan** [comment]
retest this please

**Pull request merged**